src/coreclr/nativeaot/docs/containers.md

```diff
@@ -0,0 +1,121 @@
+
+### Containerized build
+
+The following Dockerfiles demonstrate how to construct a working build environment that could be use for volume-mounted docker builds. They can be modified to use other images we provide, like for Alpine.
+
+- [Dockerfile.cross-build-x64-arm64](Dockerfile.cross-build-x64-arm64)
+- [Dockerfile.cross-build-x64-x64](Dockerfile.cross-build-x64-arm64)
+
+### x64
+
+The following pattern demonstrates the approach for x64.
+
+First build image.
+
+```bash
+$ docker build --pull -t cross-build -f Dockerfile.cross-build-x64-x64 .
+```
+
+Then build the app, with volume mounting, using the [releasesapi](https://github.com/dotnet/dotnet-docker/tree/main/samples/releasesapi) sample.
+
+```bash
+$ docker run --rm -it -v $(pwd):/source -w /source cross-build dotnet publish -o app -p:SysRoot=/crossrootfs/x64 -p:LinkerFlavor=lld releasesapi.csproj
+$  ls app
+appsettings.Development.json  nuget.config  releasesapi.dbg
+appsettings.json              releasesapi
+$ ./app/releasesapi
+info: Microsoft.Hosting.Lifetime[14]
+      Now listening on: http://localhost:5000
+info: Microsoft.Hosting.Lifetime[0]
+      Application started. Press Ctrl+C to shut down.
+info: Microsoft.Hosting.Lifetime[0]
+      Hosting environment: Production
+info: Microsoft.Hosting.Lifetime[0]
+      Content root path: /home/rich/git/dotnet-docker/samples/releasesapi
+```
+
+We can test the app in a container, again through volume mounting.
+
+```bash
+$ docker run --rm -d -v $(pwd)/app:/app -w /app -p 8000:80 mcr.microsoft.com/dotnet/runtime-deps:6.0 ./releasesapi
+219e1df9e66906531ee609b8cb9b9fa8eccef566a8682b897798d80b3905aaf5
+$ curl http://localhost:8000/healthz
+Healthy
+$ docker exec 219e1df9e66906531ee609b8cb9b9fa8eccef566a8682b897798d80b3905aaf5 cat /etc/os-release | head -n 1
+PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
+$ docker kill 219e1df9e66906531ee609b8cb9b9fa8eccef566a8682b897798d80b3905aaf5
+219e1df9e66906531ee609b8cb9b9fa8eccef566a8682b897798d80b3905aaf5
+```
+
+### Arm64
+
+Much the same pattern can be used for `x64-arm64` Dockerfile.
+
+
+```bash
+$ docker build --pull -t cross-build-arm64 -f Dockerfile.cross-build-x64-arm64 .
+$ docker run --rm -it -v $(pwd):/source -w /source cross-build-arm64 dotnet publish -a arm64 -o app-arm64 -p:SysRoot=/crossrootfs/arm64 -p:LinkerFlavor=lld releasesapi.csproj
+$ ./app-arm64/releasesapi
+-bash: ./app-arm64/releasesapi: cannot execute binary file: Exec format error
+$ file app-arm64/releasesapi
+app-arm64/releasesapi: ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, BuildID[sha1]=72212bcbd040059f1c2e6d55f640d52f7cbe2faf, stripped
+```
+
+Note the use of `-a arm64` and `-p:SysRoot=/crossrootfs/arm64` in the `dotnet publish` command. Those are required to get the build to use and generate Arm64 compatible assets.
+
+The app can then be copied to an Arm64 and it will work as demonstrated on an Apple M1 machine.
```

Comment:
```suggestion
As expected, the Arm64 app fails to run on an x64 machine. The app can then be copied to an Arm64 machine and it will work as demonstrated on an Apple M1 machine (in an Arm64 Linux container).
```